src/main/kotlin/com/dprint/services/editorservice/v4/EditorServiceV4.kt

```diff
@@ -48,24 +51,39 @@ class EditorServiceV4(private val project: Project) : EditorService {
         var status = 0
         LogUtils.info(Bundle.message("formatting.checking.can.format", filePath), project, LOGGER)
 
-        try {
-            synchronized(this) {
-                editorProcess.writeInt(CHECK_COMMAND)
-                editorProcess.writeString(filePath)
-                editorProcess.writeSuccess()
-
-                // https://github.com/dprint/dprint/blob/main/docs/editor-extension-development.md
-                // this command sequence returns 1 if the file can be formatted
-                status = editorProcess.readInt()
-                editorProcess.readAndAssertSuccess()
+        runBlocking {
+            launch {
+                try {
+                    withTimeout(CAN_FORMAT_TIMEOUT) {
+                        synchronized(editorProcess) {
```

Comment:
I am hoping this synchronisation change will fix it

Reply:
Does not fix it. I will need to try something else. I think this is happening because of the new implementation of the `AsyncDocumentFormattingService` that came recently. I might try get around this by moving the `canFormat` check outside of the backgroundable task to ensure they run on the same thread. We still have the potential of the formatting commands clashing unfortunately.

Reply:
I have asked some questions on the intellij slack to see if others have opinions on how to fix it